test/src/module/config/configTest.c

```diff
@@ -144,6 +141,30 @@ testRun(void)
         TEST_RESULT_VOID(cfgOptionSet(cfgOptConfig, cfgSourceParam, varNewStrZ("cfg")), "set option config");
         TEST_RESULT_BOOL(cfgOptionTest(cfgOptConfig), true, "option valid and value not null");
 
+        // -------------------------------------------------------------------------------------------------------------------------
+        TEST_TITLE("option groups");
+
+        TEST_RESULT_BOOL(cfgOptionGroupValid(cfgOptGrpPg), false, "pg option group is not valid");
+        TEST_RESULT_BOOL(cfgOptionGroupIndexTest(cfgOptGrpPg, 0), false, "pg option group index 0 is not set");
+        TEST_RESULT_UINT(cfgOptionGroupIndexTotal(cfgOptGrpPg), 0, "pg option group index total is 0");
+
```

Comment:
If I put the following tests here:
```
TEST_RESULT_VOID(cfgOptionValidSet(cfgOptPgPath + 5, true), "set pg5-path valid");
TEST_RESULT_VOID(cfgOptionSet(cfgOptPgPath + 5, cfgSourceDefault, VARSTRDEF("/path")), "set pg5-path");
TEST_RESULT_BOOL(cfgOptionGroupValid(cfgOptGrpPg), true, "pg option group is valid");
TEST_RESULT_BOOL(cfgOptionGroupIndexTest(cfgOptGrpPg, 5), false, "pg option group index 5 is not set");
```
Why is the group valid (`cfgOptionGroupValid(cfgOptGrpPg), true`) when no index is set?

Reply:
After our discussion, I suggest clarifying the comments on the new functions. I will submit a comment separately for where I think that could help, therefore I am resolving this conversation.